railties/test/application/generators_test.rb

```diff
@@ -172,20 +172,18 @@ def with_bare_config
       assert File.exist?(File.join(rails_root, "app/views/notifier_mailer/foo.html.erb"))
     end
 
-    test "ARGV is mutated as expected" do
-      require "#{app_path}/config/environment"
-      require "rails/command"
-      Rails::Command.const_set("APP_PATH", "rails/all")
-
-      FileUtils.cd(rails_root) do
-        ARGV = ["mailer", "notifier", "foo"]
-        Rails::Command.const_set("ARGV", ARGV)
-        quietly { Rails::Command.invoke :generate, ARGV }
-
-        assert_equal ["notifier", "foo"], ARGV
+    test "ARGV is populated" do
```

Comment:
I'm not sure if this is the best place for this test, since it's really testing `Rails::Command::GenerateCommand` behavior.  Should I create `railties/test/commands/generate_test.rb` and move this test there?